tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4740,15 +4740,22 @@ Status ConvertPad(OpConverterParams* params) {
   const auto& node_def = params->node_def;
   TF_RETURN_IF_ERROR(
       CheckInputsWeights(*params, {{"tensor", false}, {"paddings", true}}));
-  TF_RETURN_IF_ERROR(
-      AllowDataTypes(*params, {DataType::DT_FLOAT, DataType::DT_HALF}));
+  TF_RETURN_IF_ERROR(AllowDataTypes(
+      *params, {DataType::DT_FLOAT, DataType::DT_HALF, DataType::DT_INT8}));
 
   // Implement tensor binaryOp weight [channel wise] for now;
   nvinfer1::ITensor* tensor = inputs.at(0).tensor();
   const auto dims = tensor->getDimensions();
   // Restore implicit batch dimension
-  const int nb_dims = dims.nbDims + 1;
+  const int nb_dims =
+      params->use_implicit_batch ? dims.nbDims + 1 : dims.nbDims;
 
+  if (nb_dims < 4) {
+    return errors::InvalidArgument("Convertpad requires at least 4D input, at ",
+                                   node_def.name());
+    // TODO(tfeher): Support nb_dims < 4 by inserting extra dimensions to the
+    // original input.
```

Comment:
Let's move this TODO to right before the if, so it will be consistent with other TODOs  such as another on added in this PR.

Reply:
ok